src/azul/service/repository_controller.py

```diff
@@ -107,6 +107,7 @@ def search(self,
             response = self.service.search(catalog=catalog,
                                            entity_type=entity_type,
                                            file_url_func=self.file_url_func,
+                                           file_mirror_uri_func=self.mirror_service(catalog).mirror_uri,
```

Comment:
I'm not sure this indirection is necessary. For the file URL it is, because only the app/controller layer know the scheme, hostname, port and path components of that URL. But for a URI that is minted by the service, the stage should be able to reach directly into the service.